### DISCUSSION TITLE
Persisting spaCy import error: NotOpenSSLWarning: urllib3 v2.0 only supports OpenSSL 1.1.1+

### DISCUSSION THREAD

**Discussion by vrunm** [explosion/spaCy#12750] [Help: Installation] [answered]
I realize this is a recently common error, but none of the solutions I have found online helped me. I am trying to work with spaCy in Jupyter notebook and VScode on a Mac OS , but every time I try to import spacy, I get the following error:
```
NotOpenSSLWarning: urllib3 v2.0 only supports OpenSSL 1.1.1+, currently the 'ssl' module is compiled with 'LibreSSL 2.8.3'. See: https://github.com/urllib3/urllib3/issues/3020
  warnings.warn(

```
I tried

`brew install openssl@1.1`

This didn't work so I then tried to downgrade urllib3 using

`pip install urllib3==1.26.6 `

Didn't work either. I tried using a virtual environment, also didn't work. I've been scouring the web trying to find a solution but nothing seems to work.

**Comment by danieldk** [comment, marked as answer]
This is not really a spaCy issue, but an issue with the Python installation. You could try to install an older version of `urllib3`, for instance, I have a virtual environment with a recent version of spaCy 3.5 and `urllib3` `1.26.15`, which works fine:

```
pip install urllib3=1.26.15
```

If that doesn't work, please confirm with `pip show urllib3` that that version was indeed installed and what error you are getting. Another option would be to build or install a Python version that is built against OpenSSL 1.1.1 or newer.

> I tried
>
> brew install openssl@1.1

This will usually not work, because Homebrew installs openssl as a keg-only, meaning that the library is not visible, unless you explicitly link against it. Secondly, the SSL module links against libressl, which most likely has a different shared library version. Finally, the SSL module could have the library paths hardcoded (it does on my installation).

**Reply by LeshLiao** [reply]
# this work for me.
brew install openssl@1.1
pip3 install urllib3==1.26.15

**Reply by deeshank** [reply]
This worked for me as well.

**Reply by levohuutri** [reply]
This worked like a charm.

**Reply by shivamgupta97** [reply]
just a heads up for specific venv we might need to go and activate venv and then unistall existing urllib and then install the required version. There also can be issues with python permission and pip3 instal -r req.txt --user may work as a work around.

**Reply by kittykill-exe** [reply]
This worked for me too

**Reply by temtechie** [reply]
This worked for me like a magic. Thanks

**Reply by aadityaarora** [reply]
==worked for me

**Reply by berkleyxx** [reply]
pip install urllib3==1.26.15

**Reply by iwayori** [reply]
It worked, thank you very much, you saved me a lot of time!

**Reply by ryzxxn** [reply]
works like a charm. thanks.

**Reply by ramon-allink** [reply]
Thanks dude, it's really works for me too.

**Comment by amaljose01** [comment]
The error message you're encountering indicates that urllib3 version 2 requires OpenSSL 1.1.1 or newer, but your Python environment is using LibreSSL 2.8.3. This incompatibility is causing the warnings and likely the subsequent failure.

You can resolve it by following below steps, I assume the device is MacOS

**Step 1: upgrade OpenSSL**

 ```
   brew update
    brew upgrade openssl
    brew link --force openssl
```
**Step 2: Reinstall python with correct SSL**

   ```
 # Ensure pyenv is up-to-date
    brew update
    brew upgrade pyenv
    
    # Install Python with the correct OpenSSL
    env PYTHON_CONFIGURE_OPTS="--with-openssl=$(brew --prefix openssl)" pyenv install 3.9.x
    
    # Set the newly installed Python as the global version
    pyenv global 3.9.x
```

x is a place holder here, you can use like below

    `env PYTHON_CONFIGURE_OPTS="--with-openssl=$(brew --prefix openssl@1.1)" pyenv install 3.9.13
`

**Step 3: upgrade urllib3**

    `pip install --upgrade urllib3`
**Step 4: verify the installation**

   ` python -c "import ssl; print(ssl.OPENSSL_VERSION)"`

**Reply by mrtj** [reply]
I use uv. After installing and linking OpenSSL with Homebrew, setting the PYTHON_CONFIGURE_OPTS environment variable, and removing and reinstalling the virtual environment/tool with uv, the warning disappeared.

**Comment by RaffeYang** [comment]
An another way to solve this problem:
Add `export PYTHONWARNINGS=ignore` to your `.zshrc`

**Reply by dataprolet** [reply]
This is not "solving" the problem, but ignoring it.

**Comment by ratankaliani** [comment]
By default, Mac Sequoia appears to ship with a version of Python (3.9) that causes this issue.

I solved this by ensuring that the interpreter selected for my project was the Homebrew version of Python that I installed (3.13) and that the PATH was set correctly in my `.zshrc` file.

Switching to the latest Homebrew Python fixed this issue for me.
